app/services/data_sources/environment_data_source.rb

```diff
@@ -0,0 +1,24 @@
+module FastlaneCI
+  # Data source for global environment variables for fastlane.ci
+  class EnvironmentDataSource
```

Comment:
I don't love this interface. Ruby already has an interface to the environment using the `ENV` and a hash. Why not have a class extends a Hash or proxy to a hash like this? https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/base.rb#L22

Reply:
I used the same approach we use for all other model classes, e.g. https://github.com/fastlane/ci/blob/master/app/services/data_sources/user_data_source.rb, https://github.com/fastlane/ci/blob/master/app/services/config_data_sources/project_data_source.rb and https://github.com/fastlane/ci/blob/master/app/services/data_sources/notification_data_source.rb

Reply:
I suppose that is consistent with the `AbcDataSource` pattern.

But my reaction was surprise that the environment would come from something other than `ENV`, (which already has a well-established interface). Instead it's coming from a file DataSource, much like a regular config file.

Perhaps I don't understand the problem well enough.

Reply:
Ha, ok :) That makes sense. Did you have a chance to read https://github.com/fastlane/ci/blob/master/docs/use_cases/edit_global_env_variables.md, in particular the #security section: the goal is to store the environment variables for the user, similar to how you can set env variables on Heroku. This is just the basic implementation, we're gonna have to extend this class to support project and branch specific options also